lib/base_interceptor.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'gruf'
+
+module GrpcRest
+    # This is a monkey-patch that fixes an issue we were having with using the fail! method in
+    # interceptors where an active call was not instantiated yet.
+    # Basically, we overloaded this function: https://github.com/bigcommerce/gruf/blob/main/lib/gruf/errors/helpers.rb#L34
+  class BaseInterceptor < ::Gruf::Interceptors::ServerInterceptor
+    def fail!(error_code, message = 'unknown error', metadata = {})
+      raise grpc_error(error_code, message.to_s, metadata)
+    end
+
+    private
+
+    # Ported from https://github.com/flipp-oss/grpc-rest/blob/main/lib/grpc_rest.rb#L142
+    def grpc_error(error_code, message, metadata)
+      case error_code
```

Comment:
Going to see if there is any better way of doing this... Maybe we can find some way to use `BadStatus` and not break all of our tests on publication api 👀